server/src/api/FriendRoutes.ts

```diff
@@ -81,13 +88,40 @@ router.post("/", authenticateToken, async (req, res) => {
 
   const friendId = friend?.id;
 
+  // check if a friend relationship already exists
+  const existingFriend = await prisma.friend.findFirst({
+    where: {
+      OR: [
+        {
+          source_friend_id: userId,
+          destination_friend_id: friendId,
+        },
+        {
+          source_friend_id: friendId,
+          destination_friend_id: userId,
+        },
+      ],
+    },
+  });
+
+  if (existingFriend) {
+    return res.sendStatus(INVALID_REQUEST_CODE);
+  }
+
   try {
-    await prisma.friend.create({
-      data: {
-        source_friend_id: userId,
-        destination_friend_id: friendId,
-        is_accepted: false,
-      },
+    await prisma.friend.createMany({
```

Comment:
wait what isn't this going to break friend requests because you'll be able to accept your own friend request to someone else?

Reply:
ok yeah logically it doesnt make sense but when I switched it around it worked. Still looking into this 
cc @abhinit75